conda/gateways/connection/download.py

```diff
@@ -176,7 +176,7 @@ def check(target):
         target.seek(0)
         if md5 or sha256:
             checksum_type = "sha256" if sha256 else "md5"
-            checksum = sha256 if sha256 else md5
+            checksum = (sha256 if sha256 else md5).lower()
```

Comment:
A different way: `bytes.fromhex(checksum) == hasher.digest()`
An even more different way: `int(hashlib.sha256().hexdigest(),16)`

Reply:
https://github.com/dholth/conda/commit/54e01e54290b349ad569816971af10173f130c57

Reply:
Note, that if we're creating a bytes object with the `fromhex` classmethod, we'll need to account for input validation (e.g. `bytes.fromhex("foobar")` will raise a `ValueError`) while the current string based comparison is more forgiving

Reply:
Observe that in the sharded repodata CEP, checksums are transmitted as bytes and are not converted to hex.

Reply:
We can change this in rattler if it's faster that way :D Maybe we should actually type the checksum as a bytes object that is returned to Python anyways.

Reply:
here is the PR: https://github.com/mamba-org/rattler/pull/752

Reply:
That change is merged @jaimergp - not sure when we'll do the next release though as a few people are on vacation.

Reply:
Thanks @wolfv! And no worries, no rush :)

Reply:
@dholth @jezdez - added checks in bytes now. It also tests for bad strings.